### DISCUSSION TITLE
Strimzi Connect: Rest Api for delete not working properly ?

### DISCUSSION THREAD

**Discussion by Ycallaer** [strimzi/strimzi-kafka-operator#7077] [Q&A] [answered]
Hey all,
We are running different type of connectors and usually use the Rest API to interact with connect.
We currently have a connector deployed of class `io.confluent.connect.jdbc.JdbcSinkConnector` and when we try to remove the connector we use postman to send a delete request to the endpoint : https://connectserver/connectors/myawesomeconnector.

From the connect endpoint I can see that the connector gets removed, and also from the connect server logs. However the kubernetes object KafkaConnect does not get removed from the cluster and after 5 minutes it gets picked up again by connect and is registered as a connector.

Is this by design ? Do you need to remove the kubernetes object separate ? Or is this a bug ?
We are currently connect from the base image with tag `0.27.1-kafka-2.8.0`

**Comment by scholzj** [comment, marked as answer]
> However the kubernetes object KafkaConnect does not get removed from the cluster and after 5 minutes it gets picked up again by connect and is registered as a connector.

Do you really mean `KafkaConnect`? Or `KafkaConnector`?

**Reply by Ycallaer** [reply]
Hi scholzj,
Sorry I meant that the object KafkaConnector does not get removed when calling the rest api.

**Reply by scholzj** [reply]
The Connector operator uses the custom resources as a single source of truth. So when you enable it, you should not use the REST API to create, delete, modify and so on any connectors. If you do that, the operator will automatically revert your changes to make sure it matches the KafkaConnector custom resource. Really the only thing you can use the REST API for is things such as restarting connectors / tasks or monitoring them (i.e. reading the states and configurations).

So you need to decide which way you want to go. And either delete the KafkaConnector resource when you want to delete the connector. Or disable the connector operator and delete the connector using the REST API.

**Reply by Ycallaer** [reply]
Ok thanks for the explanation that makes sense.
